backend/src/stwo/prover.rs

```diff
@@ -46,16 +52,111 @@ where
-        _fixed: Arc<Vec<(String, Vec<F>)>>,
+        fixed: Arc<Vec<(String, VariablySizedColumn<F>)>>,
     ) -> Result<Self, io::Error> {
+        let split: BTreeMap<String, Analyzed<F>> = powdr_backend_utils::split_pil(&analyzed)
+            .into_iter()
+            .collect();
+
+        let preprocessed: BTreeMap<String, TableProvingKeyCollection<B, MC>> = split
+            .iter()
+            .filter_map(|(namespace, pil)| {
+                // if we have neither fixed columns nor publics, we don't need to commit to anything
+                if pil.constant_count() + pil.publics_count() == 0 {
+                    None
+                } else {
+                    let fixed_columns = machine_fixed_columns(&fixed, pil);
+                    Some((
+                        namespace.to_string(),
+                        pil.committed_polys_in_source_order()
+                            .find_map(|(s, _)| s.degree)
+                            .unwrap()
+                            .iter()
+                            .map(|size| {
+                                // get the config
+                                let config = get_config();
+
+                                let twiddles = Arc::new(B::precompute_twiddles(
+                                    CanonicCoset::new(size.ilog2() + 1 + FRI_LOG_BLOWUP as u32)
+                                        .circle_domain()
+                                        .half_coset,
+                                ));
+
+                                // Setup protocol.
+                                let prover_channel = &mut <MC as MerkleChannel>::C::default();
+                                let mut commitment_scheme =
+                                    CommitmentSchemeProver::<'_, B, MC>::new(config, &twiddles);
+
+                                let domain = CanonicCoset::new(
+                                    fixed_columns
+                                        .keys()
+                                        .next()
+                                        .map(|&first_key| first_key.ilog2())
+                                        .unwrap_or(0),
+                                )
+                                .circle_domain();
+                                // witness and constant traces need to be bit reversed
+                                let bit_reversed_fixed_columns: BTreeMap<
+                                    DegreeType,
+                                    Vec<(String, Vec<F>)>,
+                                > = fixed_columns
+                                    .iter()
+                                    .map(|(key, vec)| {
+                                        let transformed_vec: Vec<(String, Vec<F>)> = vec
+                                            .iter()
+                                            .map(|(name, slice)| {
+                                                let mut values: Vec<F> = slice.to_vec(); // Clone the slice into a Vec
```

Comment:
Cloning is not ideal here, these can be quite large. A better approach would be to introduce a type, something like

```rust
struct BitReversedColumn<'a> {
   column: &'a [F]
}
```

and implement `IntoIterator`, returning an iterator which goes through `column` in bit-reversed order.

However it seems like this is only needed for next references, which I'm not sure we should support in this PR?